src/Shared/Process/ProcessExtensions.cs

```diff
@@ -30,67 +30,7 @@ public static void KillTree(this Process process, TimeSpan timeout)
             }
             else
             {
-                var children = new HashSet<int>();
-                GetAllChildIdsUnix(pid, children, timeout);
-                foreach (var childId in children)
-                {
-                    KillProcessUnix(childId, timeout);
-                }
-                KillProcessUnix(pid, timeout);
-            }
-        }
-
-        private static void GetAllChildIdsUnix(int parentId, ISet<int> children, TimeSpan timeout)
-        {
-            try
-            {
-                RunProcessAndWaitForExit(
-                    "pgrep",
-                    $"-P {parentId}",
-                    timeout,
-                    out var stdout);
-
-                if (!string.IsNullOrEmpty(stdout))
-                {
-                    using (var reader = new StringReader(stdout))
-                    {
-                        while (true)
-                        {
-                            var text = reader.ReadLine();
-                            if (text == null)
-                            {
-                                return;
-                            }
-
-                            if (int.TryParse(text, out var id))
-                            {
-                                children.Add(id);
-                                // Recursively get the children
-                                GetAllChildIdsUnix(id, children, timeout);
-                            }
-                        }
-                    }
-                }
-            }
-            catch (Win32Exception ex) when (ex.Message.Contains("No such file or directory"))
-            {
-                // This probably means that pgrep isn't installed. Nothing to be done?
-            }
-        }
-
-        private static void KillProcessUnix(int processId, TimeSpan timeout)
-        {
-            try
-            {
-                RunProcessAndWaitForExit(
-                    "kill",
-                    $"-TERM {processId}",
-                    timeout,
-                    out var stdout);
-            }
-            catch (Win32Exception ex) when (ex.Message.Contains("No such file or directory"))
-            {
-                // This probably means that the process is already dead
+                process.Kill(entireProcessTree: true);
```

Comment:
We should probably keep `catch (Win32Exception ex) when (ex.Message.Contains("No such file or directory"))`

Reply:
I don't know if that applies to the new Process.Kill method. The catch here only applied to when pgrep wasn't installed, based on the comment.

Reply:
Ok, does look they try to avoid that error surfacing https://source.dot.net/#System.Diagnostics.Process/System/Diagnostics/Process.Unix.cs,111

Reply:
Could we change this specifically for `dotnet-watch` in 5.0? The timeout parameter is unused with this change and I'd like for us to understand how that affects things like tests before we put this in 5.0. Also, I would expect us to consistently use this API (no reason to taskkill if this guarantees the same behavior, no?)

Reply:
> Could we change this specifically for dotnet-watch in 5.0? The timeout parameter is unused with this change and I'd like for us to understand how that affects things like tests before we put this in 5.0.

Do you want this change to fix openapi as well? If so, we'd need a different solution there as well.

> Also, I would expect us to consistently use this API (no reason to taskkill if this guarantees the same behavior, no?)

If we want to minimize risk, not changing windows is probably the better move.

Reply:
After chatting with Pranav offline, we are going to only make this change for openapi and watch.

@dougbu for your awareness on openapi.

Reply:
Scratch that, the openapi portion is only for tests as well, my mistake.